ChaosMod/Effects/db/Player/PlayerWarpedControlStick.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+	Effect by Gorakh
+*/
+
+#include <stdafx.h>
+
+static int lastTick;
+
+static float horizontalOffset;
+static float verticalOffset;
+
+static void OnStart()
+{
+	horizontalOffset = g_Random.GetRandomFloat(-1.f, 1.f) * 10.f;
+	verticalOffset = g_Random.GetRandomFloat(-1.f, 1.f) / 1000000.f;
+
+	lastTick = GET_GAME_TIMER();
+}
+
+static void OnTick()
+{
+	SET_GAMEPLAY_CAM_RELATIVE_PITCH(GET_GAMEPLAY_CAM_RELATIVE_PITCH() - 1, 1.f);
+
+	for (int i = 0; i < 10; i++)
+	{
+		SET_GAMEPLAY_CAM_RELATIVE_HEADING(GET_GAMEPLAY_CAM_RELATIVE_HEADING() + (horizontalOffset / 10));
+		WAIT(0);
+	}
+
+	// TODO: Possibly offset the cursor position aswell? Windows API calls don't seem to work for that unfortunately, would probably have to find a memory address to override the position instead
```

Comment:
Have you looked at [SetCursorLocation](https://docs.fivem.net/natives/?_0xFC695459D4D0E219) it directly calls the Windows API function for [SetCursorPos](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setcursorpos). Also keep in mind that this native can alter the cursor position even when focus is not on the game window. So be careful :)

Reply:
I've already implemented this (the comment is wrong btw, it didn't work since it's parameters were pixels, while I had a float value <1, so the conversion would always end up passing an offset of 0, which of course did nothing), you're looking at an older commit, and yeah, it'll offset outside the game window (and possibly also when the game is minimized), shouldn't be too difficult to check for that though, as long as there's API calls for the window bounds and if the window is minimized it should be pretty simple.